src/rgw/rgw_sal.cc

```diff
@@ -93,213 +83,86 @@ RGWObjState::RGWObjState(const RGWObjState& rhs) : obj (rhs.obj) {
-  }
-  else if (cfg.store_name.compare("d3n") == 0) {
-    driver = new rgw::sal::RadosStore();
-    RGWRados* rados = new D3nRGWDataCache<RGWRados>;
-    dynamic_cast<rgw::sal::RadosStore*>(driver)->setRados(rados);
-    rados->set_store(static_cast<rgw::sal::RadosStore* >(driver));
-
-    if ((*rados).set_use_cache(use_cache)
-                .set_use_datacache(true)
-                .set_run_gc_thread(use_gc_thread)
-                .set_run_lc_thread(use_lc_thread)
-                .set_run_quota_threads(quota_threads)
-                .set_run_sync_thread(run_sync_thread)
-                .set_run_reshard_thread(run_reshard_thread)
-                .init_begin(cct, dpp) < 0) {
-      delete driver;
-      return nullptr;
-    }
-    if (driver->initialize(cct, dpp) < 0) {
-      delete driver;
-      return nullptr;
-    }
-    if (rados->init_complete(dpp) < 0) {
-      delete driver;
-      return nullptr;
-    }
-
-    lsubdout(cct, rgw, 1) << "rgw_d3n: rgw_d3n_l1_local_datacache_enabled=" <<
-      cct->_conf->rgw_d3n_l1_local_datacache_enabled << dendl;
-    lsubdout(cct, rgw, 1) << "rgw_d3n: rgw_d3n_l1_datacache_persistent_path='" <<
-      cct->_conf->rgw_d3n_l1_datacache_persistent_path << "'" << dendl;
-    lsubdout(cct, rgw, 1) << "rgw_d3n: rgw_d3n_l1_datacache_size=" <<
-      cct->_conf->rgw_d3n_l1_datacache_size << dendl;
-    lsubdout(cct, rgw, 1) << "rgw_d3n: rgw_d3n_l1_evict_cache_on_start=" <<
-      cct->_conf->rgw_d3n_l1_evict_cache_on_start << dendl;
-    lsubdout(cct, rgw, 1) << "rgw_d3n: rgw_d3n_l1_fadvise=" <<
-      cct->_conf->rgw_d3n_l1_fadvise << dendl;
-    lsubdout(cct, rgw, 1) << "rgw_d3n: rgw_d3n_l1_eviction_policy=" <<
-      cct->_conf->rgw_d3n_l1_eviction_policy << dendl;
-  }
-#ifdef WITH_RADOSGW_DBSTORE
-  else if (cfg.store_name.compare("dbstore") == 0) {
-    driver = newDBStore(cct);
-
-    if ((*(rgw::sal::DBStore*)driver).set_run_lc_thread(use_lc_thread)
-                                    .initialize(cct, dpp) < 0) {
-      delete driver;
-      return nullptr;
-    }
-  }
-#endif
-
-#ifdef WITH_RADOSGW_MOTR
-  else if (cfg.store_name.compare("motr") == 0) {
-    driver = newMotrStore(cct);
-    if (driver == nullptr) {
-      ldpp_dout(dpp, 0) << "newMotrStore() failed!" << dendl;
-      return driver;
-    }
-    ((rgw::sal::MotrStore *)driver)->init_metadata_cache(dpp, cct);
-
-    return store;
-  }
-#endif
-
-#ifdef WITH_RADOSGW_DAOS
-  else if (cfg.store_name.compare("daos") == 0) {
-    driver = newDaosStore(cct);
-    if (driver == nullptr) {
-      ldpp_dout(dpp, 0) << "newDaosStore() failed!" << dendl;
-      return driver;
-    }
-    int ret = driver->initialize(cct, dpp);
-    if (ret != 0) {
-      ldpp_dout(dpp, 20) << "ERROR: store->initialize() failed: " << ret << dendl;
-      delete driver;
-      return nullptr;
-    }
+  auto dlname = fmt::format("/usr/lib64/ceph/librgw_sal_{}.so", cfg.store_name);
+  auto altdlname = fmt::format("./lib/librgw_sal_{}.so", cfg.store_name);
```

Comment:
i think this path should be relative to `plugin_dir` from ceph.conf. vstart.sh sets that to build/lib so we shouldn't need the ./lib/ fallback

Reply:
will be fixed in the next push